eval.c

```diff
@@ -236,7 +236,8 @@ sexp sexp_extend_synclo_env (sexp ctx, sexp env) {
       sexp_env_renames(e2) = sexp_env_renames(e1);
 #endif
     }
-    sexp_env_parent(e2) = sexp_context_env(ctx);
+    // ISSUE_296 - Dereference of null pointer, stored line 231
+    if (e2) { sexp_env_parent(e2) = sexp_context_env(ctx); }
```

Comment:
Looking through the Static Analyzer report, it seems to be complaining about the case where env may be NULL (probably impossible) in the function call.  In that case the for loop never iterates, e1 and e2 are both NULL, and the sexp_env_parent(e2) is dereferencing a NULL pointer.

Put in what I believe is an OOM check to silence the warning, which probably isn't an accurate error for this case:

```c
    if (!e2) { return sexp_global(ctx, SEXP_G_OOM_ERROR); }
    sexp_env_parent(e2) = sexp_context_env(ctx);
```

As far as checking all possible OOM cases, is there a reason why this couldn't be done in the allocator (`sexp_alloc_type`) and have it return the SEXP exception rather than at the result sites?